### PR TITLE
YD-590 Register caches explicitly

### PR DESCRIPTION
**Pull request by Bert-R**
The Hazelcast caches don't update metrics unless explicitly registered.

### CODE REVIEW

File: core/src/main/java/nu/yona/server/CacheConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package nu.yona.server;
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/CacheConfiguration.java]: We usually put the copyright comment above the `package` line

File: core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java [hunk 2]
```diff
@@ -63,19 +65,35 @@
...
 	@Autowired
 	private ActivityCategoryRepository repository;
 
+	@Autowired(required = false)
+	private CacheConfiguration cacheConfiguration;
+
 	private static final Logger logger = LoggerFactory.getLogger(ActivityCategoryService.class);
 
+	@PostConstruct
+	public void register()
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java]: Suggestion: `registerCacheForMetrics`

File: core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java [hunk 3]
```diff
@@ -63,19 +65,35 @@
...
+	}
+
 	@Transactional
 	public ActivityCategoryDto getActivityCategory(UUID id)
 	{
 		ActivityCategory activityCategoryEntity = getEntityById(id);
 		return ActivityCategoryDto.createInstance(activityCategoryEntity);
 	}
 
-	@Cacheable(value = "activityCategorySet", key = "'instance'")
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java]: This fixes a bug I assume?

File: core/src/main/java/nu/yona/server/CacheConfiguration.java (hunk 1 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/CacheConfiguration.java]: True. I now corrected ``CoreConfiguration`` too.

File: core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java (hunk 2 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java]: Makes sense

File: core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java (hunk 3 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/goals/service/ActivityCategoryService.java]: I'm not sure whether I would call it a bug, but the ``@CacheConfig`` annotation named the cache ``activityCategories`` while this annotation here named it ``activityCategorySet``, and the last one won. Now the cache is named ``activityCategories``.

**Comment by Bert-R** [comment]
Pushed an update

**Pull request merged**